### PR TITLE
Align stack to 16 bytes to avoid SIGBUS

### LINKED ISSUE #193: ObjC runtime tests hit SIGBUS with go 1.22rc1
Hey!

As seen in #191, the `objc_runtime_darwin_test.go` tests appear to crash with `go 1.22rc1`.
I haven't been able to diagnose exactly what causes this issue just yet (and also am not entirely sure where to start from).

Since go1.22 is expected to release some time in February, it is probably a good idea to investigate this and address the problem.

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
Closes #193

### CODE REVIEW

**Comment by TotallyGamerJet** [comment]
Is there something wrong with the 1.18 server? It was running fine in https://github.com/TotallyGamerJet/purego/commit/06e7a14b260bc48aa20a69f68e33c552ccbffa60

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

I'll cherry-pick this to 0.5.

**Comment by TotallyGamerJet** [comment]
> LGTM, thanks!

I don't want to merge it while the 1.18 test hasn't run

**Comment by hajimehoshi** [comment]
> I don't want to merge it while the 1.18 test hasn't run

I restaretd the CI test.

File: sys_amd64.s [hunk 1]
```diff
@@ -154,8 +154,7 @@ TEXT callbackasm1(SB), NOSPLIT|NOFRAME, $0
 
 	POP_REGS_HOST_TO_ABI0()
 
-	MOVQ  0(SP), R10 // get the SP back
-	ADDQ  $8, SP
+    POPQ R10 // get the SP back
```
Comment (hajimehoshi) [inline sys_amd64.s:157]: tab?

Comment (TotallyGamerJet) [reply sys_amd64.s]: yeah I fixed it. Sorry about making another change. I realized I could save two instructions by actually pushing and popping

**Review by hajimehoshi — APPROVED** [review]
Still LGTM

**Pull request merged**